exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/InternalBatch.java

```diff
@@ -88,4 +91,7 @@ public void clear() {
     return container.getValueAccessorById(clazz, fieldIds);
   }
 
+  public FragmentContext getContext() {
+    return incoming.getContext();
```

Comment:
OK, so we hold onto the incoming batch just to get the FragmentContext. If so, please pass the FragmentContext into the constructor of this class rather than passing in the (irrelevant) incoming batch.